File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result = {{1}};
        vector<int>temp;
        for(int i = 2; i < numRows + 1; i++) {
            temp.clear();
            for(int j = 0; j < i; j++) {
                if(j == 0 or j == i-1) {
                    temp.push_back(1);
                }
                else {
                    temp.push_back(result[i-2][j-1] + result[i-2][j]);
                }
            }
            result.push_back(temp);
        }
        return result;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        rows = [[1]]
        prev = rows[-1]
        for _ in range(numRows - 1):
            row = [1]
            for i, n in enumerate(prev[:-1]):
                row.append(n + prev[i+1])
            row.append(1)
            rows.append(row)
            prev = row
        return rows